### PR TITLE
`hf 14a apdu` improvement

### PR DESCRIPTION
**Pull request by merlokk**
added:

1. print APDU (if it can be decoded)
2. construct extended and normal size apdu

### CODE REVIEW

File: client/emv/apduinfo.h [hunk 1]
```diff
@@ -31,4 +32,28 @@ typedef struct {
 const APDUCode *GetAPDUCode(uint8_t sw1, uint8_t sw2);
 const char *GetAPDUCodeDescription(uint8_t sw1, uint8_t sw2);
 
+typedef struct {
+    uint8_t cla;
+    uint8_t ins;
+    uint8_t p1;
+    uint8_t p2;
+    uint8_t lc[3];
+} __attribute__((packed)) ExtAPDUHeader;
```
Comment (iceman1001) [inline client/emv/apduinfo.h]: Don't we have a define PACKED for this?

File: client/emv/apduinfo.h [hunk 2]
```diff
@@ -31,4 +32,28 @@ typedef struct {
...
+    uint8_t cla;
+    uint8_t ins;
+    uint8_t p1;
+    uint8_t p2;
+    uint16_t lc;
+    uint8_t *data;
+    uint32_t le;
+    bool extended_apdu;
+    uint8_t case_type;
+} __attribute__((packed)) APDUStruct;
```
Comment (iceman1001) [inline client/emv/apduinfo.h]: See above comment

**Comment by doegox** [comment]
and changelog... ;)

File: client/emv/apduinfo.h (hunk 1 above)
Comment (merlokk) [reply client/emv/apduinfo.h]: yes, but i see it in
`include\hitag.h` and `include\pm3_cmd.h`

have added it to `util.h`

**Comment by doegox** [comment]
thanks!

**Comment by iceman1001** [comment]
I think we already have the PACKED defined somewhere before????

**Comment by merlokk** [comment]
here `include\hitag.h` and `include\pm3_cmd.h`

maybe ill add include\proxmark.h?
and add there?

**Comment by iceman1001** [comment]
sounds like a good suggestion

**Comment by merlokk** [comment]
just need to choose right name...
or maybe  `common_client.h`?
there is another proxmark3.h

**Comment by iceman1001** [comment]
just need to choose right header file :)  it must be one in common folder since we use the PACKED define on both sides.

**Comment by merlokk** [comment]
in include folder there is another common.h, but it for arm. almost all headers there for arm.

maybe lets client lay in folder client...

**Comment by iceman1001** [comment]
as I said,  we use that define on both device and client.

**Comment by merlokk** [comment]
ill create new one and put there min max and this

**Comment by doegox** [comment]
common.h is already used both in client & arm

**Comment by doegox** [comment]
while you're at it, please replace the ` __attribute__((__packed__))` still present in `client/elf.h`, `armsrc/Standalone/hf_young.c`, `include/mifare.h` and `include/common.h`

**Comment by merlokk** [comment]
Ok.
I see that common.h is place there was added many arm related stuff. Try to get rid of that

**Comment by doegox** [comment]
ok, yes include files are a bit messy, rationalization is welcomed :)

**Pull request merged**